src/jit/morph.cpp

```diff
@@ -9550,8 +9550,7 @@ GenTreePtr          Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* ma
         {
             GenTreePtr effOp1 = op1->gtEffectiveVal();
             
-            noway_assert( (effOp1->gtOper == GT_CNS_INT) &&
-                         ((effOp1->gtIntCon.gtIconVal == 0) || (effOp1->gtIntCon.gtIconVal == 1)) );
+            noway_assert(effOp1->IsZero() || effOp1->IsIOne());
```

Comment:
Here too the condition has become more conservative and allows float/long zero and long one as permissible values.  The right condition here is

effOp1->IsIntZero() || effOp1->IsIntOne()